version: '3'

services:
  nginx:
    image: nginx:latest
    container_name: nginx
    ports:
      - "8080:80"
    volumes:
      - ./nginx/default.conf:/etc/nginx/conf.d/default.conf
    networks:
      - node-network
    depends_on:
      - app

  app:
    build:
      context: node
    container_name: app
    volumes:
      - ./node:/usr/src/app
      - /usr/src/app/node_modules
    working_dir: /usr/src/app
    tty: true
    ports:
      - "3000:3000"
    environment:
      - DB_HOST=db
      - DB_USER=root
      - DB_PASSWORD=root
      - DB_NAME=nodedb
    networks:
      - node-network      
    depends_on:
      db:
        condition: service_healthy

  db:
    image: mysql:5.7
    command: --innodb-use-native-aio=0
    container_name: db
    restart: always
    tty: true
    environment:
      - MYSQL_DATABASE=nodedb
      - MYSQL_ROOT_PASSWORD=root
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "root", "-p$$MYSQL_ROOT_PASSWORD"]
      interval: 5s
      timeout: 10s
      retries: 3
    networks:
      - node-network

    volumes:
      - ./mysql:/var/lib/mysql

networks:
  node-network:
    driver: bridge